File: guessnumber.py
import random 
def game():  
    num = random.randint(1, 100) 
    total = 0 
    n = 0
    while True: 
        n = int(input("Guess the number")) 
        total += 1 
        if n < num: 
            print("No, too low") 
        elif n > num: 
            print("No, too much") 
        else: 
            print(f' Yes! You win with {total} gueses') 
            break 
game()

File: guessnumber.lua
function game()
	num = math.random(100)
	n = 0
	while n ~= num do
		print("guess number:")
		n = io.read("*number")
		if n < num then
			print("no, too low")
		elseif n > num then
			print("no, too much")
		end
	end
end

game()